File: hse_learning/cpp_lang/contest_Y/num_A.cpp
#include <iostream>
#include <vector>
#include <cmath>

int main() {
    // n - этажи
    // m - квартиры
    // x - высота y - ширина в квартире
    int n, m, x, y;
    std::cin >> n >> m >> x >> y;

    std::vector<std::string> windows(n * x);
    for (int i = 0; i < n * x; i++) {
        std::cin >> windows[i];
    }

    int ans = 0;

    for (int floor = 0; floor < n; floor++) {
        for (int k = 0; k < m; k++) {
            int light = 0;

            for (int i = 0; i < x; ++i) {
                for (int j = 0; j < y; ++j) {
                    if (windows[floor * x + i][k * y + j] == 'X') {
                        light++;
                    }
                }
            }

            if (light >= std::ceil((x * y) / 2.0)) {
                ans++;
            }
        }
    }

    std::cout << ans << std::endl;

    return 0;
}

File: hse_learning/go/contest_Y/num_A.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	var n, m, x, y int
	fmt.Scan(&n, &m, &x, &y)

	reader := bufio.NewReader(os.Stdin)
	windows := make([][]byte, n*x)

	for i := 0; i < n*x; i++ {
		line, _ := reader.ReadString('\n')
		windows[i] = []byte(line[:len(line)-1])
	}

	ans := 0

	for floor := 0; floor < n; floor++ {
		startRow := floor * x
		for apt := 0; apt < m; apt++ {
			startCol := apt * y
			light := 0

			for i := 0; i < x; i++ {
				for j := 0; j < y; j++ {
					ch := windows[startRow+i][startCol+j]
					if ch == 'X' {
						light++
					}
				}
			}

			if light*2 >= x*y {
				ans++
			}
		}
	}

	fmt.Println(ans)
}
